Formula/infer.rb

```diff
@@ -29,33 +35,42 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    opamroot = buildpath/"opamroot"
-    opamroot.mkpath
+    ENV.prepend_path "PKG_CONFIG_PATH", Formula["sqlite"].opt_lib/"pkgconfig"
+
+    # opamroot = HOMEBREW_CACHE/"opam"
+    opamroot = libexec/"opam"
     ENV["OPAMROOT"] = opamroot
     ENV["OPAMYES"] = "1"
 
-    # do not attempt to use the clang in facebook-clang-plugins/ as it hasn't been built yet
-    ENV["INFER_CONFIGURE_OPTS"] = "--prefix=#{prefix} --without-fcp-clang"
+    # explicitly build the clang before infer's configure
+    ENV["INFER_CONFIGURE_OPTS"] = "--prefix=#{prefix} --with-fcp-clang"
 
     llvm_args = %w[
       -DLLVM_INCLUDE_DOCS=OFF
       -DLLVM_INSTALL_UTILS=OFF
-      -DLLVM_TARGETS_TO_BUILD=all
       -DLIBOMP_ARCH=x86_64
-      -DLLVM_BUILD_EXTERNAL_COMPILER_RT=ON
-      -DLLVM_BUILD_LLVM_DYLIB=ON
     ]
 
-    system "opam", "init", "--no-setup"
-    ocaml_version = File.read("build-infer.sh").match(/OCAML_VERSION_DEFAULT=\"([^\"]+)\"/)[1]
-    ocaml_version_number = ocaml_version.split("+", 2)[0]
-    inreplace "#{opamroot}/compilers/#{ocaml_version_number}/#{ocaml_version}/#{ocaml_version}.comp",
-      '["./configure"', '["./configure" "-no-graph"'
+    pathfix_lines = [
+      "export SDKROOT=#{MacOS.sdk_path}",
+      "eval $(opam env)"
+    ]
+
     # so that `infer --version` reports a release version number
-    inreplace "infer/src/base/Version.ml.in", "let is_release = is_yes \"@IS_RELEASE_TREE@\"", "let is_release = true"
+    inreplace "infer/src/base/Version.ml.in", "@IS_RELEASE_TREE@", "yes"
+
     inreplace "facebook-clang-plugins/clang/setup.sh", "CMAKE_ARGS=(", "CMAKE_ARGS=(\n  " + llvm_args.join("\n  ")
-    system "./build-infer.sh", "all", "--yes"
-    system "opam", "config", "exec", "--switch=infer-#{ocaml_version}", "--", "make", "install"
+
+    # setup opam (disable opam sandboxing because homebrew is sandboxed already)
+    inreplace "build-infer.sh", "--no-setup", "--no-setup --disable-sandboxing"
```

Comment:
You explain what this is doing, but *why* is it doing this? Is it actually needed? Can we get upstream to come up with something saner than this hacky install script?

Reply:
So I know for opam, initializing a repo by default uses sandboxing. This is fine for a __user__ install (typically at `~/.opam`) but it results it a permissions-denied error if you try to initialize an opam repo inside the homebrew-sandboxed build environment.

<details>
<summary>partial log from homebrew build failure <code>01.build-infer.sh</code></summary>
<pre><code>
[ERROR] The compilation of ocaml-variants failed at "/usr/local/Cellar/infer/0.16.0/libexec/opam/opam-init/hooks/sandbox.sh buil
d ./configure -prefix /usr/local/Cellar/infer/0.16.0/libexec/opam/ocaml-variants.4.07.1+flambda -with-debug-runtime -flambda -cc
 cc -aspp cc -c".

\#=== ERROR while compiling ocaml-variants.4.07.1+flambda ======================#
\# context              2.0.4 | macos/x86_64 |  | https://opam.ocaml.org#b878563b
\# path                 /usr/local/Cellar/infer/0.16.0/libexec/opam/ocaml-variants.4.07.1+flambda/.opam-switch/build/ocaml-varian
ts.4.07.1+flambda
\# command              /usr/local/Cellar/infer/0.16.0/libexec/opam/opam-init/hooks/sandbox.sh build ./configure -prefix /usr/local/Cellar/infer/0.16.0/libexec/opam/ocaml-variants.4.07.1+flambda -with-debug-runtime -flambda -cc cc -aspp cc -c
\# exit-code            71
\# env-file             /usr/local/Cellar/infer/0.16.0/libexec/opam/log/ocaml-variants-55607-d8a4ae.env
\# output-file          /usr/local/Cellar/infer/0.16.0/libexec/opam/log/ocaml-variants-55607-d8a4ae.out
\### output ###
\# sandbox-exec: sandbox_apply_container: Operation not permitted
</code></pre>
</details>